panther_hardware_interfaces/src/panther_system.cpp

```diff
@@ -0,0 +1,519 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_hardware_interfaces/panther_system.hpp>
+
+#include <rclcpp/logging.hpp>
+
+#include <hardware_interface/types/hardware_interface_type_values.hpp>
+
+#include <panther_hardware_interfaces/utils.hpp>
+
+// todo: add user variable to the script that will trigger DOUT4 instead of safety stop
+
+namespace panther_hardware_interfaces
+{
+
+using namespace std::literals;
+
+void PantherSystem::CheckJointSize() const
+{
+  if (info_.joints.size() != kJointsSize) {
+    throw std::runtime_error(
+      "Wrong number of joints defined: "s + std::to_string(info_.joints.size()) + ", "s +
+      std::to_string(kJointsSize) + "expected."s);
+  }
+}
+
+void PantherSystem::SortJointNames()
+{
+  // Sort joints names - later hw_states and hw_commands are accessed by static indexes, so it
+  // is necessary to make sure that joints are in specific order and order of definitions in URDF
+  // doesn't matter
+  for (std::size_t i = 0; i < kJointsSize; i++) {
+    for (std::size_t j = 0; j < kJointsSize; j++) {
+      if (info_.joints[j].name.find(joint_order_[i]) != std::string::npos) {
+        joints_names_sorted_[i] = info_.joints[j].name;
+      }
+    }
+  }
```

Comment:
```suggestion
  for (std::size_t i = 0; i < kJointsSize; i++) {
    for (std::size_t j = 0; j < kJointsSize; j++) {
      if (info_.joints[j].name.find(joint_order_[i]) != std::string::npos) {
        joints_names_sorted_[i] = info_.joints[j].name;
        break;
      }
    }
  }
```
no point in checking the rest if correct one is found. 
The other thing I'm concerned is what if joint name will accidentally contain joint_order key? For example: `fl_frword_wheel`, `fr_frword_wheel`, `rl_frword_wheel`, `rr_frword_wheel`. In that case sorting wont work

Reply:
that's a good point, maybe I will add checking if there is exactly one reference to the key.

Reply:
Still you have to remember not to put any of joint_order keys in joint name. Maybe we assume that key should be `<x>fr_` where x can be empty or `_` or `/` in case of namespaces or additional prefixes? it's lot more complex I know but a bit safer.